### PR TITLE
feat(nav): added tokens, new design spec

### LINKED ISSUE #6000: Consume tokens in: Navigation

### PR DESCRIPTION
**Pull request by mattnolting**
Fixes #6000

### CODE REVIEW

**Comment by srambach** [comment]
In pages, you can see that we've got a large spacer on the inline-start of the page sidebar AND on the navigation. I think the padding should be removed from the page sidebar, WDYT?

**Comment by mattnolting** [comment]
> In pages, you can see that we've got a large spacer on the inline-start of the page sidebar AND on the navigation. I think the padding should be removed from the page sidebar, WDYT?

@srambach Yep, I agree 👍

File: src/patternfly/components/Nav/nav-item.hbs
Comment (srambach) [inline src/patternfly/components/Nav/nav-item.hbs:20]: Just curious about the reason to put the partial-block inside a conditional

**Comment by andrew-ronaldson** [comment]
Agree with the comments above on the spacer

File: src/patternfly/components/Nav/nav.scss
Comment (srambach) [inline src/patternfly/components/Nav/nav.scss:67]: Here you have the width and color mixed up. Also, I think the color should be `--pf-t--global--border--color--default`.

File: src/patternfly/components/Nav/nav.scss
Comment (srambach) [inline src/patternfly/components/Nav/nav.scss:44]: Can you leave a `// TODO` here about using a transition token once available?

File: src/patternfly/components/Nav/nav.scss
Comment (srambach) [inline src/patternfly/components/Nav/nav.scss:106]: Can you leave a `// TODO` here about using a transition token once available?

**Review by srambach — COMMENTED** [review]
Couple of minor things, but looks really good! Maybe you can walk us through the handlebars helpers sometime?

**Comment by lboehling** [comment]
this is looking great 🎉, thank you @mattnolting!! a few of comments --

- I agree with the spacing comments above. can you remove the left and right padding from main nav sections, so it just has the top/bottom padding? like this 
Screenshot 2023-11-15 at 10 22 11 AM

- Can you remove the right padding from nested the nav section so it extends full width on the right like this?
Screenshot 2023-11-15 at 10 23 02 AM

- Can we bump up group header text to body-md (14px)? In the designs I also had it set to small (12px) but i think that it reads better larger now that i see it in build! I'll need to update the design for this too. 

- The horizontal sub nav is showing within the page content area. Should it show below the masthead or should I update design to make it look better when placed in the page section?
Screenshot 2023-11-15 at 10 28 42 AM

File: src/patternfly/components/Nav/nav-item.hbs
Comment (mattnolting) [reply src/patternfly/components/Nav/nav-item.hbs:20]: Great question, the conditional allows for additional content in the `nav__item`. Without the conditional, partial content is required, which then requires a closing tag 

`{{#> nav-item nav-link--text='my-content'}}{{/nav-item}}` 

vs 

`{{> nav-item nav-link--text='my-content'}}`)

**Comment by mattnolting** [comment]
@lboehling 

> I agree with the spacing comments above. can you remove the left and right padding from main nav sections, so it just has the top/bottom padding? like this

Done. Thanks @srambach 👍 

> Can you remove the right padding from nested the nav section so it extends full width on the right like this?

Done. 

Screenshot 2023-11-16 at 11 38 03 AM

> Can we bump up group header text to body-md (14px)? In the designs I also had it set to small (12px) but i think that it reads better larger now that i see it in build! I'll need to update the design for this too.

Done.

Screenshot 2023-11-16 at 11 38 46 AM

> The horizontal sub nav is showing within the page content area. Should it show below the masthead or should I update design to make it look better when placed in the page section?

Screenshot 2023-11-16 at 11 26 45 AM

@lboehling I think it would be great to see both examples. The example you're referring to is controlled by `page` rather than `nav`, so we'd need to consider how main/subnav is presented within page context. I'd also like to see `nav/subnav` is the masthead in both mobile and desktop formats. 


Screenshot 2023-11-16 at 11 47 25 AM

Screenshot 2023-11-16 at 11 43 33 AM

Should `scroll-buttons` have interaction states (hover/focus/active) other than text color? 

Should `scroll-buttons` have a `box-shadow` when the list exceeds the available space? 

Something like this: 

Screenshot 2023-11-16 at 11 55 43 AM

**Review by andrew-ronaldson — COMMENTED** [review]
Screenshot 2023-12-16 at 8 41 00 AM
Not sure if this was addressed above but the subnav content has a padding-inline-start of 24px but the designs are 16px.

**Review by andrew-ronaldson — APPROVED** [review]
Looks great, thanks!

File: src/patternfly/components/Nav/examples/Navigation.md
Comment (srambach) [inline src/patternfly/components/Nav/examples/Navigation.md:37]:
```suggestion
### New Expandable
```
Hierarchy in the headings here - but also, what should these really be called? (probably not "New" right?)

**Review by srambach — COMMENTED** [review]
The carets seem to be broken in the preview?

File: src/patternfly/components/Nav/nav.scss
Comment (thatblindgeye) [inline src/patternfly/components/Nav/nav.scss:2]: Other components have been updated to root + just the component selector here:

```suggestion
:root,
.#{$nav} {
```

Do we want to shift to utilizing `:where` here?

File: src/patternfly/components/Nav/nav.scss
Comment (thatblindgeye) [inline src/patternfly/components/Nav/nav.scss:67]: This would be worth adding back in, as right now the focus outline gets cut off in horizontal examples:

image

File: src/patternfly/components/Nav/examples/Navigation.md
Comment (mattnolting) [reply src/patternfly/components/Nav/examples/Navigation.md:37]: Ah, yes. Those examples are for demo purposes. They'll be removed before merge.

File: src/patternfly/components/Nav/nav.scss
Comment (mattnolting) [reply src/patternfly/components/Nav/nav.scss:2]: The core team talked about this earlier this week. Using `:where()` here resolves some variables conflicts while allowing `vars()` to be updated/themed without wrapping in the `.pf-v5-c-component` selector.

**Review by thatblindgeye — APPROVED** [review]

File: src/patternfly/components/Button/button.scss
Comment (mcoker) [inline src/patternfly/components/Button/button.scss:285]: Curious what this is needed for? Looks like the horizontal nav arrow icons align OK without it.

File: src/patternfly/components/Button/button.scss
Comment (mcoker) [inline src/patternfly/components/Button/button.scss:630]: This works but so far we tried not to have RTL modifiers for everything that accepts an icon, and instead use either the mixins in core or react has been using `<Icon shouldMirrorRTL />`. On the react side I'd expect this to be setup as

```
<Button variant="plain">
  <Icon shouldMirrorRTL><AngleRightIcon /></Icon>
</Button>
```

I'd probably do the same on the core side. You should always be able to pass the icon component to a button as the `icon` prop or as a child of a plain button. That also prevents any confusion around the global class `.pf-m-mirror-inline-rtl` and this local `.pf-m-RTL-icon-static` class, which are named differently, but do the same thing.

**Review by mcoker — COMMENTED** [review]
LGTM, the only thing I'd like to look at before merging is the button enhancement to have its own class to mirror/flip its icon, since that's adding a feature to the react button. IIRC, last we spoke about it we wanted to try and use `<Icon shouldMirrorRTL>` anywhere there isn't a core class we can target to apply a mixin and you can pass the icon to something.

**Review by mcoker — CHANGES_REQUESTED** [review]
Oh forgot to leave this comment - the new button vars you've introduced are not initially defined in the button component. Not sure if that's on purpose or not, but at a minimum the border-radius vars need to be added back since the button has lost its rounded corners https://patternfly-pr-6033.surge.sh/components/button

**Review by mcoker — APPROVED** [review]

**Pull request merged**
